src/tasks.js

```diff
@@ -118,8 +122,22 @@ async function processNextTaskRound(task) {
   const transactions = Array.from(
     task.transactions.slice(nextI, nextI + task.batchSize)
   )
+  const modifiedTask = await TaskModel.findById(task._id)
+
+  if (modifiedTask.status === 'Paused') {
+    logger.info(
+      `Processing of task ${task._id} paused. Remaining transactions = ${task.remainingTransactions}`
+    )
+    TaskModel.findByIdAndUpdate({_id: task._id}, {
+      remainingTransactions: task.remainingTransactions
+    })
+
+    return
+  }
 
   const promises = transactions.map(transaction => {
+    task.remainingTransactions--
```

Comment:
Better to decrement once the promise is resolve ? wdyt ?

Reply:
Decrementing after will result in the rerunning process being slow. This is refactoring that we did to reduce the length of rerunning when we do bulk reruns for data recovery